src/botservice/azext_bot/custom.py

```diff
@@ -0,0 +1,466 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+from knack.prompting import prompt_y_n
+from knack.util import CLIError
+from knack.log import get_logger
+from azext_bot.botservice import AzureBotService
+from azext_bot.botservice.models import Bot, BotProperties,sku, BotChannel
+from azure.cli.command_modules.appservice.custom import enable_zip_deploy, config_source_control, get_app_settings, _get_site_credential, _get_scm_url
+from azure.cli.command_modules.resource.custom import deploy_arm_template
```

Comment:
Depending on other command modules like this is potentially brittle.
The command modules you import have no idea you're using them in this way so this could break the extension in the future.

It's okay to leave it as is since you mentioned you plan to move in to the CLI so it will be less of an issue then but still not recommended.

Reply:
that is true. longterm, should i talk to the other cli module teams to make sure that they continue to support these functions?

Reply:
In general, your extension/module should not rely on other modules so you shouldn't have these imports at all. You can either copy the functionality into your module or we could move them into the core so that a common implementation is available.

Reply:
either of the 2 funcitonalities are acceptable to me. i am thinking of making the changes in another PR though since i dont want to break anything this close to build conference. Is it ok to get this checked in and onto the index.json and then taking the dependency off (by copying the code over)?